Modules/Loadable/Segmentations/Widgets/qMRMLSegmentationGeometryWidget.cxx

```diff
@@ -0,0 +1,803 @@
+
+// Segmentations includes
+#include "qMRMLSegmentationGeometryWidget.h"
+
+#include "ui_qMRMLSegmentationGeometryWidget.h"
+
+#include "vtkSlicerSegmentationsModuleLogic.h"
+#include "vtkMRMLSegmentationNode.h"
+
+// SegmentationCore includes
+#include "vtkSegmentation.h"
+#include "vtkSegmentationConverter.h"
+#include "vtkOrientedImageData.h"
+#include "vtkOrientedImageDataResample.h"
+#include "vtkCalculateOversamplingFactor.h"
+
+// MRML includes
+#include "vtkMRMLScalarVolumeNode.h"
+#include "vtkMRMLTransformNode.h"
+#include "vtkMRMLAnnotationROINode.h"
+#include "vtkMRMLModelNode.h"
+
+// VTK includes
+#include <vtkWeakPointer.h>
+#include <vtkMatrix4x4.h>
+#include <vtkTransform.h>
+#include <vtkGeneralTransform.h>
+#include <vtkAddonMathUtilities.h>
+
+// Qt includes
+#include <QDebug>
+
+//-----------------------------------------------------------------------------
+class qMRMLSegmentationGeometryWidgetPrivate: public Ui_qMRMLSegmentationGeometryWidget
+{
+  Q_DECLARE_PUBLIC(qMRMLSegmentationGeometryWidget);
+
+protected:
+  qMRMLSegmentationGeometryWidget* const q_ptr;
+public:
+  qMRMLSegmentationGeometryWidgetPrivate(qMRMLSegmentationGeometryWidget& object);
+  virtual ~qMRMLSegmentationGeometryWidgetPrivate();
+  void init();
+
+  /// Calculate output geometry into \sa GeometryImageData based on selection
+  /// \return Error message. Empty when successful
+  QString calculateOutputGeometry();
+
+  /// Fill geometry info section from geometry image data containing geometry information \sa GeometryImageData
+  void updateGeometryWidgets();
+
+  /// Reset geometry image data \sa GeometryImageData
+  void resetGeometryImageData();
+
+  /// Determine if source node is a non-empty segmentation with a labelmap master
+  bool isSourceASegmentationWithBinaryLabelmapMaster();
+
+  /// Determine if input segmentation is non-empty, and with binary labelmap master
+  bool inputSegmentationCanBeResampled();
+
+  // Match the axes of the input labelmap to the axes of the transformed source node.
+  // Use this calculated permutation for updating spacing widget from geometry and interpreting spacing input
+  void matchInputAndSourceAxes(vtkMRMLNode* sourceNode);
+
+public:
+  /// Segmentation MRML node
+  vtkWeakPointer<vtkMRMLSegmentationNode> SegmentationNode;
+
+  /// Oriented image data containing output geometry
+  vtkOrientedImageData* GeometryImageData;
+
+  /// Source to input axes mapping \sa matchInputAndSourceAxes
+  int InputAxisIndexForSourceAxis[3];
+  /// Inverse permutation of \sa InputAxisIndexForSourceAxis
+  int SourceAxisIndexForInputAxis[3];
+
+  /// Flag indicating whether editing is enabled in the widget
+  bool EditEnabled;
+};
+
+//-----------------------------------------------------------------------------
+qMRMLSegmentationGeometryWidgetPrivate::qMRMLSegmentationGeometryWidgetPrivate(qMRMLSegmentationGeometryWidget& object)
+  : q_ptr(&object)
+  , InputAxisIndexForSourceAxis{ 0, 1, 2 }
```

Comment:
This may require Cxx11-compatible compilers, which we did not want to require in Slicer-4.10 (just right after releasing 4.10).